### ISSUE TITLE
Build tool plugin is not seeing target dependencies when there is exactly one

### ISSUE THREAD

**Issue by DougGregor** [swiftlang/swift-package-manager#8060]
### Is it reproducible with SwiftPM command-line tools: `swift build`, `swift test`, `swift package` etc?

- [X] Confirmed reproduction steps with SwiftPM CLI. The description text _must_ include reproduction steps with either of command-line SwiftPM commands, `swift build`, `swift test`, `swift package` etc.

### Description

I have a build tool plugin that looks at the dependencies of the target it is applied to. For example:

```swift
        .target(
            name: "JavaMath",
            dependencies: [
              .product(name: "JavaKit", package: "swift-java"),
              // Note: Uncomment the following to unbreak test.swift
              // .product(name: "JavaKitJar", package: "swift-java"),
            ],
            swiftSettings: [
              .unsafeFlags(["-I\(javaIncludePath)", "-I\(javaPlatformIncludePath)"])
            ],
            plugins: [
              .plugin(name: "Java2SwiftPlugin", package: "swift-java"),
            ]
        ),
```

The Java2SwiftPlugin doesn't see the dependency of JavaMath on JavaKit. However, if I uncomment the second line (to add the JavaKitJar dependency), the Java2SwiftPlugin sees both the JavaKit and JavaKitJar dependencies correctly. It seems to only ever happen when there is exactly one dependency, it is missing.

### Expected behavior

The plugin should see the target dependency even when there is only one.

### Actual behavior

The dependency is missing when there is a single dependency; all dependencies are visible if there is more than one.

### Steps to reproduce

1. `git clone https://github.com/DougGregor/swift-java-deps`
2. `cd swift-java-deps`
3. `swift build`

The build will fail. If you uncomment the marked line in `Package.swift`, the build will succeed because the Java2SwiftPlugin is correctly seeing the dependency.

### Swift Package Manager version/commit hash

6.0

### Swift & OS version (output of `swift --version ; uname -a`)

```
swift-driver version: 1.115 Apple Swift version 6.0 (swiftlang-6.0.0.9.10 clang-1600.0.26.2)
Target: arm64-apple-macosx15.0
```

**Comment by dschaefer2** [comment]
@DougGregor just to confirm, were you getting errors like:

`.../swift-java-deps/.build/plugins/outputs/swift-java-deps/JavaMath/destination/Java2SwiftPlugin/generated/RoundingMode.swift:20:13: error: value of type 'RoundingMode' has no member 'equals'`

**Comment by dschaefer2** [comment]
If I change the Package.swift to only depend on JavaKitJar (which is really JavaKitReflection), it works. So I don't think the number of dependencies are a factor. Unless you are getting a totally different error than the one I posted above.

**Comment by DougGregor** [comment]
That's the right error. The first indication of failure is actually a warning earlier on:

```
warning: Unable to translate 'java.math.RoundingMode' method 'equals': Java class 'java.lang.Object' has not been translated into Swift
```

which is because the build tool isn't seeing the JavaKit target. If I go ahead and dump the JavaMath target's dependencies as the build tool sees it, "JavaKit" doesn't show up.

**Comment by DougGregor** [comment]
You're right that the number of dependencies isn't a factor. Something about the JavaKit product/target specifically is tripping it up: if I dump the dependencies of the target in the build tool, we don't see a direct dependency on JavaKit! But if I swap in JavaKitJar, we see a direct dependency on it and an indirect one on JavaKit.

**Comment by dschaefer2** [comment]
Fixed a bug on the main branch where plugin dependencies were sneaking into the build planning. With that, swift-java-deps builds correctly.

However, I'm still seeing the reported behaviour on the 6.0 branch. I want to understand that before assuming it's fixed on main.

**Comment by dschaefer2** [comment]
Hmm, finally back to this. I'm seeing the duplicate module map on this package now, even with my "fix". Debugging further.

**Comment by dschaefer2** [comment]
D'oh. The fix was broken. Reapplying https://github.com/swiftlang/swift-package-manager/pull/8498.

BTW, with this fix, I was able to build swift-java-deps.

**Comment by dschaefer2** [comment]
This is working with SwiftPM's native build system. Leaving this open because it is not working with `--build-system swiftbuild`.

```
error: Clang dependency scanner failure: While building module 'JavaRuntime' imported from JavaRuntime-d514b82d.input:1:
In file included from <module-includes>:1:
/Users/dschaefer2/swift/swift-java-deps/.build/checkouts/swift-java/Sources/JavaRuntime/include/JavaRuntime.h:18:10: fatal error: 'jni.h' file not found
JavaRuntime-d514b82d.input:1:1: fatal error: could not build module 'JavaRuntime'
error: Unable to find module dependency: 'JavaRuntime'
```
